1.preprocessing_data/scripts/1.qc_report.py

```diff
@@ -0,0 +1,164 @@
+
+# In[3]:
+
+
+# Read in CSV files
+outlier_df <- read.csv(path_to_outlier_results)
+qc_df <- read.csv(path_to_cp_qc_results)
+
+# Add 'Metadata_Well' and 'Metadata_Site' columns using filename from 'FileName_OrigActin'
+qc_df <- qc_df %>%
+  mutate(Metadata_Well = gsub("_", "", str_extract(FileName_OrigActin, "_([^f]+)")),
+         Metadata_Site = str_extract(FileName_OrigActin, "f.{2}"))
+         
+head(qc_df)
+
+
+# ## Generate data frame removing poor quality sites and counting the number of sites left per well
+
+# In[4]:
+
+
+# Load in platemap and add cell type to include in plate map figure
+platemap_df <- read.csv("../metadata/localhost231120090001_platemap.csv")
+
+# Find the number of FOVs that are being removed due to being poor images
+unique_combinations <- outlier_df[c('Metadata_Well', 'Metadata_Site')] %>% distinct()
+cat("The number of image sets being removed is", nrow(unique_combinations), "out of 960 image sets\n")
+
+# Remove the image sets from the main data frame to generate bar chart
+result_df <- anti_join(qc_df, unique_combinations, by = c('Metadata_Well', 'Metadata_Site'))
+
+# Group by 'Metadata_Well' and count the occurrences of 'Metadata_Site'
+counts_per_well <- result_df %>% group_by(Metadata_Well) %>% summarise(count = n())
+
+# Select only relevant columns
+platemap_subset <- platemap_df[, c("well_position", "cell_type")]
+
+# Add cell type to counts per well
+counts_per_well <- merge(counts_per_well, platemap_subset, 
+                         by.x = "Metadata_Well", by.y = "well_position", all.x = TRUE)
+
+dim(counts_per_well)
+head(counts_per_well)
+
+
+# ## Generate bar chart to show how many FOVs were processed after QC
+
+# In[5]:
+
+
+# Extract the first letter from 'Metadata_Well' to create a color palette
+unique_starting_letters <- unique(substr(counts_per_well$Metadata_Well, 1, 1))
+
+# Create a color palette based on unique starting letters
+color_palette <- colorRampPalette(brewer.pal(8, "Set1"))(length(unique_starting_letters))
+
+# Map each unique starting letter to a color in a named vector
+color_dict <- setNames(color_palette, unique_starting_letters)
+
+# Create a new column 'Color' in the data frame based on the starting letter
+counts_per_well$Color <- color_dict[substr(counts_per_well$Metadata_Well, 1, 1)]
+
+# Increase the figure size to extend the chart horizontally
+options(repr.plot.width=14, repr.plot.height=6)
+
+# Create a bar chart using ggplot2 with the 'Color' column for colors
+fov_chart <- ggplot(counts_per_well, aes(x = Metadata_Well, y = count, fill = Color)) +
+  geom_bar(stat = 'identity') +
+  labs(x = 'Metadata_Well', y = 'Number of FOVs', title = 'Count of FOVs per Well for plate localhost231120090001') +
+  scale_fill_identity() +
+  theme_bw() +
+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
+
+# Save plot to qc_figures
+ggsave(
+        barchart_output_file,
+        fov_chart,
+        dpi = 500,
+        height = 6,
+        width = 10
+    )
```

Comment:
Consider viewing and saving these plots with the same width and height by defining the width and heights as a variable first.

```suggestion
width <- 14
height <- 6
options(repr.plot.width=width, repr.plot.height=height)

# Create a bar chart using ggplot2 with the 'Color' column for colors
fov_chart <- ggplot(counts_per_well, aes(x = Metadata_Well, y = count, fill = Color)) +
  geom_bar(stat = 'identity') +
  labs(x = 'Metadata_Well', y = 'Number of FOVs', title = 'Count of FOVs per Well for plate localhost231120090001') +
  scale_fill_identity() +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Save plot to qc_figures
ggsave(
        barchart_output_file,
        fov_chart,
        dpi = 500,
        height = height,
        width = width
    )
```

Reply:
I like this change and I will try to keep this format in the future. Thanks!